cmd/clusterctl/client/tree/discovery.go

```diff
@@ -151,21 +260,44 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 		}
 	}
 
-	// Handles orphan machines.
-	if len(machineMap) < len(machinesList.Items) {
-		other := VirtualObject(cluster.Namespace, "OtherGroup", "Other")
-		tree.Add(workers, other)
+	return nil
+}
 
-		for i := range machinesList.Items {
-			m := &machinesList.Items[i]
-			if _, ok := machineMap[m.Name]; ok {
-				continue
-			}
-			addMachineFunc(other, m)
+func addMachinePoolsToObjectTree(cluster *clusterv1.Cluster, workers *unstructured.Unstructured, machinePoolList *expv1.MachinePoolList, tree *ObjectTree, options DiscoverOptions) {
+	for i := range machinePoolList.Items {
+		mp := &machinePoolList.Items[i]
+		addOpts := make([]AddObjectOption, 0)
+		if !options.ShowTemplates {
+			addOpts = append(addOpts, GroupingObject(true))
+		}
```

Comment:
I'm not sure I understand why are we mixing up flags effects here.
Ideally the flags should not overlap